supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystem.scala

```diff
@@ -224,4 +190,45 @@ trait SharedFileSystem extends PathFactory {
       case x => Success(x)
     }
   }
+
+  /**
+    * Return a `Success` result if the file has already been localized, otherwise `Failure`.
+    */
+  private def localizePathAlreadyLocalized(originalPath: File, executionPath: File): Try[Unit] = {
+    if (executionPath.exists) Success(()) else Failure(new RuntimeException(s"$originalPath doesn't exists"))
+  }
+
+  private def localizePathViaCopy(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
+    val executionTmpPath = pathPlusSuffix(executionPath, ".tmp")
+    val result = Try(originalPath.copyTo(executionTmpPath, overwrite = true).moveTo(executionPath, overwrite = true)).void
+    if (result.isFailure) logger.warn(s"Localization via copy has failed: ${result.failed.get.getMessage}")
+    result
+  }
+
+  private def localizePathViaHardLink(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
+    val result = Try(executionPath.linkTo(originalPath, symbolic = false)).void
```

Comment:
@geoffjentry please check localize... functions and you will realize about required logging changes.